specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2020-03-01/cosmos-db.json

```diff
@@ -0,0 +1,7248 @@
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/accountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/DatabaseAccountListReadOnlyKeysResult"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DocumentDB/databaseAccounts/{accountName}/regenerateKey": {
+      "post": {
+        "operationId": "DatabaseAccounts_RegenerateKey",
+        "x-ms-examples": {
+          "CosmosDBDatabaseAccountRegenerateKey": {
+            "$ref": "./examples/CosmosDBDatabaseAccountRegenerateKey.json"
+          }
+        },
+        "description": "Regenerates an access key for the specified Azure Cosmos DB database account.",
+        "x-ms-long-running-operation": true,
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/accountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "name": "keyToRegenerate",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/DatabaseAccountRegenerateKeyParameters"
+            },
+            "description": "The name of the key to regenerate."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK"
+          },
+          "202": {
+            "description": "Accepted. The regenerate key operation will complete asynchronously."
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.DocumentDB/databaseAccountNames/{accountName}": {
+      "head": {
+        "operationId": "DatabaseAccounts_CheckNameExists",
+        "x-ms-examples": {
+          "CosmosDBDatabaseAccountCheckNameExists": {
+            "$ref": "./examples/CosmosDBDatabaseAccountCheckNameExists.json"
+          }
+        },
+        "description": "Checks that the Azure Cosmos DB account name already exists. A valid account name may contain only lowercase letters, numbers, and the '-' character, and must be between 3 and 50 characters.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/accountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The account name is valid but is already in use."
+          },
+          "404": {
```

Comment:
Is 404  the right error code in this case?

Reply:
That is what the API returns currently for all other versions. Its not changing in this version update.